lti_consumer/plugin/views.py

```diff
@@ -101,6 +102,42 @@ def has_block_access(user, block, course_key):
     return course_access and block_access
 
 
+def parse_custom_parameters(block):
+    """
+    Parse XBlock custom_parameters field values.
+
+    This allows us to transform the list items on the custom_parameters field into a dictionary of custom parameters
+    that can be used with the LtiConsumer1p3 set_custom_parameters method, it will also resolve any dynamic custom
+    parameter present on the field.
+
+    Args:
+        block: xblock Object to get custom_parameters from
+
+    Returns:
+        dict: Parsed custom_parameters dictionary
+
+    Raises:
+        LtiError: If custom_parameter cannot be parsed.
+    """
+    result = {}
+
+    for p in block.custom_parameters:
+        try:
+            k, v = [i.strip() for i in p.split('=', 1)]
+
+            # Resolve dynamic custom parameter value.
+            if re.search(r'\$\{\ *\w+\ *\}', v):
+                v = resolve_custom_parameter_template(block, v)
+
+            result.update({k: v})
+        except ValueError as exc:
+            raise LtiError(
+                f"Could not parse custom parameter: {p}. Should be 'x=y' string."
```

Comment:
Can you add another clearer example in this message?

Reply:
@anfbermudezme I updated the message but I'm not sure what else to add to make it clearer